webapi/src/main/scala/org/knora/webapi/responders/v2/SearchResponderV2.scala

```diff
@@ -2545,6 +2545,283 @@ class SearchResponderV2 extends ResponderWithStandoffV2 {
+
+                    /*println("+++++++++")
+
+                    println("graph pattern check for " + mainResIri)
+
+                    println("expected dependent resources: " + expectedDependentResources)
+
+                    println("all expected dependent resources present: " + allDependentResources)
+
+                    println("given dependent resources " + resAndValueObjIris.resourceIris)
+
+                    println("expected value objs: " + expectedValueObjects)
+
+                    println("given value objs: " + resAndValueObjIris.valueObjectIris)
+
+                    println("all expected value objects present: " + allValueObjects)*/
+
+                    if (allDependentResources && allValueObjects) {
+                        // sufficient permissions, include the main resource and its values
+                        acc + (mainResIri -> values)
+                    } else {
+                        // insufficient permissions, skip the resource
+                        acc
+                    }
+            }
+
+        }
+
+        /**
+          * Given the results of the main query, filters out all values that the user did not ask for in the input query,
+          * i.e that are not present in its CONSTRUCT clause.
+          *
+          * @param queryResultsWithFullGraphPattern        results with full graph pattern (that user has sufficient permissions on).
+          * @param valueObjectVarsAndIrisPerMainResource   value object variables and their Iris per main resource.
+          * @param allResourceVariablesFromTypeInspection  all variables representing resources.
+          * @param dependentResourceIrisFromTypeInspection Iris of dependent resources used in the input query.
+          * @param transformer                             the transformer that was used to turn the input query into the prequery.
+          * @param typeInspectionResult                    results of type inspection of the input query.
+          * @return results with only the values the user asked for in the input query's CONSTRUCT clause.
+          */
+        def getRequestedValuesFromResultsWithFullGraphPattern(queryResultsWithFullGraphPattern: Map[IRI, ConstructResponseUtilV2.ResourceWithValueRdfData],
+                                          valueObjectVarsAndIrisPerMainResource: Map[IRI, Map[QueryVariable, Set[IRI]]],
+                                          allResourceVariablesFromTypeInspection: Set[QueryVariable],
+                                          dependentResourceIrisFromTypeInspection: Set[IRI],
+                                          transformer: NonTriplestoreSpecificConstructToSelectTransformer,
+                                          typeInspectionResult: GravsearchTypeInspectionResult): Map[IRI, ConstructResponseUtilV2.ResourceWithValueRdfData] = {
+
+            // sort out those value objects that the user did not ask for in the input query's CONSTRUCT clause
```

Comment:
did or did not ask for?

Reply:
clarified in 
https://github.com/dhlab-basel/Knora/pull/1028/commits/cbb0e22807efbbfb12415babf5dac0d4e3197cb2